update-script.js

```diff
@@ -0,0 +1,141 @@
+ *
+ * @param {KintoClient} client KintoClient instance
+ * @param {string} bucket Name of the Remote Settings bucket
+ * @param {Object} newRecord Object containing the updated related realms object
+ * @param {string} newRecord.id ID from the current related realms object from the Remote Settings server
+ * @param {string[][]} newRecord.relatedRealms Updated related realms array from GitHub
+ */
+const updateRecord = async (client, bucket, newRecord) => {
+  await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).updateRecord(newRecord);
+  const postServerData = await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).getData();
+  const setDataObject = {
+    status: "to-review",
+    last_modified: postServerData.last_modified
+  };
+  await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).setData(setDataObject, {patch: true});
+  console.log(`Found new records, committing changes to ${WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION} collection.`);
+};
+
+/**
+ * Creates a new record in Remote Settings if there are no records in the WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION
+ *
+ * @param {KintoClient} client
+ * @param {string} bucket
+ */
+const createRecord = async (client, bucket) => {
+  let githubRecords = await getSourceRecords();
+  const result = await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).createRecord({
+    relatedRealms: githubRecords
+  });
+  const postServerData = await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).getData();
+  await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).setData({status: "to-review", last_modified: postServerData.last_modified}, {patch: true});
+  console.log(`Added new record to ${WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION}`, result);
+};
+
+const printSuccessMessage = () => {
+  console.log("Script finished successfully!");
+}
+
+/**
+ * The runner for the script.
+ * 
+ * @return {Number} 0 for success, 1 for failure.
+ */
+const main = async () => {
+  if (FX_RS_WRITER_USER === "" || FX_RS_WRITER_PASS === "") {
+    console.error("No username or password set, quitting!");
+    return 1;
+  }
+  const bucket = "main-workspace";
+  const secretString = `${FX_RS_WRITER_USER}:${FX_RS_WRITER_PASS}`;
+  try {
+    const client = new KintoClient(SERVER_ADDRESS, {
+      headers: {
+        Authorization: "Basic " + btoa(secretString)
+      }
+    });
+
+    let records = await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).listRecords();
+    let data = records.data;
+
+    // If there are existing records in the collection, we need to update instead of creating new records
+    if (data.length) {
+      let currentRelatedRealms = data[0].relatedRealms;
+      let id = data[0].id;
+      let areNewRecords = false;
+      let githubRecords = await getSourceRecords();
```

Comment:
I think you could move that above and remove `getSourceRecords()` from `createRecord()`

Reply:
I'm not sure I understand. Are you suggesting moving L102-105 to before the `if (data.length)` statement?

Reply:
Well, it would be more consistent if `createRecord()` wouldn't be responsible of fetching the source, since `updateRecord()` isn't. 
Moving the call `getSourceRecords()` above all this and passing it to `createRecord()` would be easier to follow IMO

Reply:
I think I eventually got to your main point on this comment, happy to adjust though.